src/openforms/submissions/api/renderers.py

```diff
@@ -21,3 +21,13 @@ class FileRenderer(renderers.BaseRenderer):
 
     def render(self, data, media_type=None, renderer_context=None):
         return data
+
+
+class JSONOrPlainTextRenderer(renderers.JSONRenderer):
```

Comment:
shouldn't this inherit from the `CamelCaseJSONRenderer`?

I'm also thinking if it wouldn't be cleaner to implement `def finalize_response(self, request, response, *args, **kwargs):` on the particular view and look at `response.status_code`, and if that's `HTTP 400`, then change the `request.accepted_renderer` and `request.accepted_media_type`. This then allows you to implement a Plain Text renderer and a regular JSON renderer without these sort of nasty `isinstance` checks, since tecnically the string `"foo"` can still be regular JSON:


```py
>>> import json
>>> json.dumps("foo")
'"foo"'
```

Reply:
I applied this suggestion in the PR.